hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/index/functional/BaseHoodieIndexClient.java

```diff
@@ -61,4 +62,9 @@ public void register(HoodieTableMetaClient metaClient, String indexName, String
    * Create a functional index.
    */
   public abstract void create(HoodieTableMetaClient metaClient, String indexName, String indexType, Map<String, Map<String, String>> columns, Map<String, String> options);
+
+  /**
+   * Drop an index. By default, ignore drop if index does not exist.
+   */
+  public abstract void drop(HoodieTableMetaClient metaClient, String indexName, MetadataPartitionType metadataPartitionType, boolean ignoreIfNotExists);
```

Comment:
isn't the 3rd arg is needs fixes?
for sec and functional, the partition type could be same, but it could refer to diff indexes depending on the col we built the index on.

Reply:
oh I see you are getting the index name too. 
can we add java docs then.

Reply:
also, can we compute the metadataPartitionType internally. 
```
val metadataPartitionType = MetadataPartitionType.fromPartitionPath(indexName)
```

we don't need the 3rd arg then.

Reply:
+1, could you consolidate `indexName` and `metadataPartitionType` into one argument?

Reply:
Is there a case we need to set different `ignoreIfNotExists`?  If not, can we remove the argument too?

Reply:
I have removed `metadataPartitionType` argument. On `ignoreIfNotExists`, we support two types of DROP INDEX statements:
```
DROP INDEX <index_name> ON <table_name>;
DROP INDEX IF EXISTS <index_name> ON <table_name>;
```
In the first case `ignoreIfNotExists` is false while true in the latter case. So, in the sql prompt, the error will be thrown if ignore is set to false so that user knows that the index already doesn't exist. Just need this to support different sql flavors.